frontend/pages/events.tsx

```diff
@@ -123,37 +125,55 @@ const ViewContext = createContext<
   [
     option: EventsViewOption,
     setOption?: Dispatch<SetStateAction<EventsViewOption>>,
+    showSyncModal?: () => void,
   ]
 >([EventsViewOption.CALENDAR])
 
 /**
  * Component used to display a method to switch between the event grid and calendar views.
  */
-const EventsViewSwitcher = ({ viewOption, setViewOption }): ReactElement => (
-  <div className="buttons has-addons mt-0 mb-0">
+const EventsViewToolbar = ({
+  viewOption,
+  setViewOption,
+  showSyncModal,
+}): ReactElement => (
+  <div style={{ display: 'flex' }}>
+    <div className="buttons has-addons mt-0 mb-0">
+      <button
+        id="event-view-list"
+        className={`button is-medium ${
+          viewOption === EventsViewOption.LIST ? 'is-selected is-info' : ''
+        }`}
+        aria-label="switch to grid view"
+        onClick={() => {
+          setViewOption(EventsViewOption.LIST)
+        }}
+      >
+        <Icon name="grid" alt="grid view" />
+      </button>
+      <button
+        id="event-view-calendar"
+        className={`button is-medium ${
+          viewOption === EventsViewOption.CALENDAR ? 'is-selected is-info' : ''
+        }`}
+        aria-label="switch to calendar view"
+        onClick={() => {
+          setViewOption(EventsViewOption.CALENDAR)
+        }}
+      >
+        <Icon name="calendar" alt="calendar view" />
+      </button>
+    </div>
+    {/* margins are for matching vertical alignments with grouped buttons */}
     <button
-      id="event-view-list"
-      className={`button is-medium ${
-        viewOption === EventsViewOption.LIST ? 'is-selected is-info' : ''
-      }`}
-      aria-label="switch to grid view"
       onClick={() => {
-        setViewOption(EventsViewOption.LIST)
+        showSyncModal && showSyncModal()
       }}
+      className="button is-medium"
+      style={{ marginLeft: '20px', marginBottom: '8px' }}
     >
-      <Icon name="grid" alt="grid view" />
-    </button>
-    <button
-      id="event-view-calendar"
-      className={`button is-medium ${
-        viewOption === EventsViewOption.CALENDAR ? 'is-selected is-info' : ''
-      }`}
-      aria-label="switch to calendar view"
-      onClick={() => {
-        setViewOption(EventsViewOption.CALENDAR)
-      }}
-    >
-      <Icon name="calendar" alt="calendar view" />
+      <Icon name="refresh" style={{ marginRight: '10px' }} />
+      Export
```

Comment:
Not sure if this is intentional or you just want a space between the icon and the text, you can do `{' '}` to force React to acknowledge a space here.